### PR TITLE
Improve developer README

### PR DESCRIPTION
**Pull request by gwenwindflower**
This is a:
- [x] documentation update
- [ ] bug fix with no breaking changes
- [ ] new functionality
- [ ] a breaking change

## Description & motivation

I found, even as an experienced analytics engineer, being new to package development that the developer README for contributors in `integration_tests/README.md` could use some expansion and clarifying to be more welcoming. I've attempted to do this.

In doing this I also wanted to clarify the `.env` / `profiles.yml` flow to be simpler, as that was my first initial source of confusion. This PR proposes that we:

- store the CI `profiles.yml` in the root of the project so it's more discoverable and saves us a step in CI
- get rid of the `.env/[adapter].env` files, which are run the risk of somebody filling in and committing their credentials, and instead list the env vars needed for each adapter in the expanded README.

## Checklist
- [n/a] This code is associated with an issue which has been triaged and [accepted for development](https://docs.getdbt.com/docs/contributing/oss-expectations#pull-requests). 
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [n/a] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: integration_tests/README.md [hunk 1]
```diff
@@ -1,141 +1,197 @@
...
+5. [Commit your changes and open a pull request](#commit-your-changes-and-open-a-pull-request)
+
+## Overview
 
 ### Prerequisites
-- python3
-- Docker (optional)
+
+- [python3](https://www.python.org/)
+- [make](<https://en.wikipedia.org/wiki/Make_(software)>) (Optional, but highly recommended for better development experience)
```
Comment (joellabes) [inline integration_tests/README.md:22]: How do I install make if I don't have it already? the wiki page didn't seem to clarify

**Review by joellabes — APPROVED** [review]
Approved at a vibes level as opposed to a line-by-line review of the code! I like your plan and justification, and believe in your implementation

File: integration_tests/README.md (hunk 1 above)
Comment (gwenwindflower) [reply integration_tests/README.md:22]: great question! this was never included in the original guide so i didn't even think to add it. it's included with most C-focused language tool chains, so on MacOS when you install xcode-command-line tools (or whatever the exact name of the package is called, it's installed with that) -- but we should definitely...._make_ this clearer, will add!

**Comment by gwenwindflower** [comment]
> Approved at a vibes level as opposed to a line-by-line review of the code! I like your plan and justification, and believe in your implementation

ty 💗 -- i really wish the markdown had already been auto-linted/formatted as it makes it hard to see exactly what the intentional changes were, as opposed to all the formatting that was automatically applied by Prettier updating the markdown, but now it will be clean moving forward (everybody using VSCode will have the same auto-formatting/linting of the Markdown as this, so should be pretty universal).

i'll add some details re `make` installation and get this in. 🙏🏻

**Pull request merged**